iree/compiler/Dialect/VM/IR/test/arithmetic_folding.mlir

```diff
@@ -131,6 +131,23 @@ vm.module @mul_i32_folds {
 
 // -----
 
+// CHECK-LABEL: @mul_mul_i32_folds
+vm.module @mul_mul_i32_folds {
+  // CHECK-LABEL: @mul_mul_i32_const
+  vm.func @mul_mul_i32_const(%arg0 : i32) -> i32 {
+    // CHECK: %c40 = vm.const.i32 40 : i32
+    %c4 = vm.const.i32 4 : i32
+    %c10 = vm.const.i32 10 : i32
+    // CHECK: %0 = vm.mul.i32 %arg0, %c40 : i32
+    %0 = vm.mul.i32 %arg0, %c4 : i32
+    %1 = vm.mul.i32 %0, %c10 : i32
```

Comment:
What about the case where these arguments are flipped?
`%1 = vm.mul.i32 %c10, %0 : i32`
Should the new code handle that case as well? Or canonicalize so if one arg is a constant it always appears second?

Reply:
In MLIR all constant operands get moved to the right-hand side of commutative ops like mul, so we only need to handle the one case of `[lhs value], [rhs constant]`. This isn't the case for the add+sub folding, where both operands are values and may be in either order.

(the implicit constants-on-rhs threw me at first as it's not really discoverable, so in general I still try to write things to handle both when it's not too ugly as there's little cost and much improved readability)